workflow/source.go

```diff
@@ -138,6 +148,37 @@ func (d Document) IsRemote() bool {
 	return getFileStatus(d.Location) == fileStatusRemote
 }
 
+func (d Document) IsSpeakeasyRegistry() bool {
+	return strings.Contains(d.Location, "registry.speakeasyapi.dev")
+}
+
+func (d Document) ParseSpeakeasyRegistryReference() *SpeakeasyRegistryDocument {
+	if !d.IsSpeakeasyRegistry() {
+		return nil
+	}
+
+	registryDocument := &SpeakeasyRegistryDocument{}
+	subPath := strings.Split(d.Location, "registry.speakeasyapi.dev/")[1]
+	components := strings.Split(subPath, "/")
+	registryDocument.OrgSlug = components[0]
+	registryDocument.WorkspaceSlug = components[1]
+
+	endComponent := strings.SplitN(components[2], ":", 2)
+	registryDocument.NamespaceName = endComponent[0]
+	reference := endComponent[1]
```

Comment:
What do we do in the case that there isn't a digest or tag reference? currently this will cause a panic because we're accessing `endComponent[1]` without checking the split result. My opinion is if there's no reference specified we default to `"latest"`.

Reply:
Good point. I'm good with the latest default